specification/visualstudio/resource-manager/Microsoft.VisualStudio/2014-04-01-preview/examples/CreateProjectResource.json

```diff
@@ -0,0 +1,28 @@
+﻿{
+  "parameters": {
+    "body": {
+        "id": "/subscriptions/0de7f055-dbea-498d-8e9e-da287eedca90/resourceGroups/VS-Example-Group/providers/Microsoft.VisualStudio/account/ExampleAccount/project/ExampleProject",
+        "name": "ExampleProject",
+        "type": "Microsoft.VisualStudio/account/project",
+        "location": "Central US",
+        "tags": {},
+        "properties": {
+          "ProcessTemplateId": "6B724908-EF14-45CF-84F8-768B5384DA45",
```

Comment:
you are correct that there isn't a way to do this as of today. For the future, I hear that there is a proposal to expand the scope of [x-ms-enum](https://github.com/Azure/azure-rest-api-specs/blob/dce4da0d748565efd2ab97a43d0683c2979a974a/documentation/swagger-extensions.md#x-ms-enum) to accommodate this kind of friendly name assignment.